### ISSUE TITLE
Sharp segfault

### ISSUE THREAD

**Issue by SoprachevAK** [oven-sh/bun#3218]
### What version of Bun is running?

0.6.7 and 0.6.8 canary

### What platform is your computer?

Darwin 22.5.0 x86_64 i386

### What steps can reproduce the bug?

1. `bun init`
2. `bun add sharp`
3. Install sharp module addons: `cd node_modules/sharp/ && npm run install`
4.  index.ts: 
```ts
// index.ts
import sharp from 'sharp'; //OR const sharp = require('sharp');

const image = sharp({
  create: {
    width: 128,
    height: 128,
    channels: 4,
    background: { r: 0, g: 0, b: 0, alpha: 0.5 }
  }
})

for (let i = 0; i < 100; i++) {
  const buf = await image.resize({ width: 16, height: 16 })
    .raw()
    .toBuffer()

  console.log(i, buf.byteLength);
  await Bun.sleep(1); // | 10 | 100
}
```
5. run `bun index.ts`

### What is the expected behavior?

print 100 output lines
```bash
0 1024
1 1024
2 1024
<...>
99 1024
```

### What do you see instead?

Segmentation fault around **third** step
```bash
0 1024
1 1024
2 1024
<...>
Segmentation fault: 11
```

### Additional information

1. the bug is reproduced regardless of the include method (`import` or `require`)
2. reproduced depends of `image size` and `sleep`. For example:
- `1024x1024` and `sleep(1)` works correct
- `1024x1024` and `sleep(10)` segfault after 65 step
- `128x128` without `sleep` segfault after 4 step
- `128x128` and `sleep(10)` segfault after 2 step

3. if you open the file from the disk `sharp('input.png')` bug is still reproduced
4. if you save image to file, works correct and save all 100 images
```ts
import sharp from 'sharp';

const image = sharp({
  create: { width: 128, height: 128, channels: 4, background: { r: 0, g: 0, b: 0, alpha: 0.5 } }
})

for (let i = 0; i < 100; i++) {
  const buf = await image.resize({ width: 16, height: 16 })
    .png()
    .toFile(`./output/${i}.png`);
    
  await Bun.sleep(1); //OR without
}

```

**Comment by SoprachevAK** [comment]
From [discord](https://discord.com/channels/876711213126520882/1113918029097603233) 
> Jarred — 01.06.2023 22:53
> Smells like GC issue with NAPI

**Comment by samuelgja** [comment]
I cannot get sharp lib run at all.

For me your steps gives this, so sharp seems to be still not compatible with m1 mac or? 

bun index.ts:
```
error: 
Something went wrong installing the "sharp" module

Cannot find module "../build/Release/sharp-darwin-arm64v8.node" 
```


bun -v **0.6.9**
Darwin 22.4.0 arm64 arm

**Comment by birkskyum** [comment]
For the upcoming wasm version of sharp, these worker threads are important - more info here https://blog.stackblitz.com/posts/bringing-sharp-to-wasm-and-webcontainers/

- #901
- https://github.com/oven-sh/bun/pull/3923

**Comment by paperclover** [comment]
Fixed in 0.8.0 / canary
